python/ray/data/read_api.py

```diff
@@ -1047,14 +1047,33 @@ def read_tfrecords(
            length  width    species
         0     5.1    3.5  b'setosa'
 
+        We can also read compressed TFRecord files which uses one of the
+        `compression type supported by Arrow <https://arrow.apache.org/docs/python/generated/pyarrow.CompressedInputStream.html>`:
+
+        >>> compressed_path = os.path.join(tempfile.gettempdir(), "data_compressed.tfrecords")
+        >>> options = tf.io.TFRecordOptions(compression_type="ZLIB") # "GZIP" also supported by TensorFlow
```

Comment:
nit: given we read the gzip file, it's kind of weird that we wrote w/ `zlib` but read w/ `gzip`, can we change it to:

```
>>> options = tf.io.TFRecordOptions(compression_type="GZIP") # "ZLIB" also supported by TensorFlow
```